getMultimedias.js

```diff
@@ -70,14 +75,31 @@
   const numOfTitles = numOfTitlesElem.textContent.split(' ')[0];
 
   const loadMore = () => {
-    // const scrollAmount = document.documentElement.scrollHeight - document.documentElement.clientHeight;
+	// const scrollAmount = document.documentElement.scrollHeight - document.documentElement.clientHeight;
     const scrollAmount = 212 * 25; // each element's height is close to 212px and to make sure we load all the images we scroll 25 elements by 25
     document.documentElement.scrollTop += scrollAmount;
-
-    const allTitles = document.querySelectorAll('.ipc-metadata-list-summary-item');
+	
+    const numOfTitlesAsInt = parseInt(numOfTitles);  
+    const maxScrollPos = ((numOfTitlesAsInt - 3) * 207) + 450; // Set max scroll position based on num of titles, avg element's height 207px and the lenghth from page top to 1st element in the list 450px 
+    // Define event listener function
+    function handleScroll() {
```

Comment:
The idea here is putting max position from top in pixels that the scroll bar should not exceed, the max pos calculated based on number of  titles in the page, the average height of the title and the distance between the top and 1st movie in the page.

Reply:
I thought about this when I first updated the code, but we can not ensure that this is going to be covering different base font sizes of users that they set from their browsers. We can assume base px value is going to be 16px, but I don't think the effort for those calculations will worth it. In summary your `masScrollPos` calculation will be false for some users.

Let's stick with the current safe one, and one flawed approach I have currently is the `isInView` check, but I will update it.

Reply:
I made another code that's much better and doesn't depend on fonts. I'll push it soon.

Reply:
What I tried to say is that the optimal solution should not need anything related with the sizing. If the last element is in dom, then we can just scroll into it and done. In your solution you are creating an event listener and removing it with a setTimeout which is creating unnecessary complexity and hard to read.

Reply:
ok, as u like, it's your code.

Reply:
But, the good thing about this code that it keeps the viewport above this black footer while scrolling 

![ScreenShot_20240603221826](https://github.com/AlkTheOrg/imdb-watchlist-randomizer/assets/110132520/2879534c-95ac-458c-a66c-c8765a241562)

 as it's really annoying because it prevents the user from seeing where the scrolling reached at the moment. With this black footer, The user won't know how far or near the scrolling is from the end until it reaches the end.

Also, this code shouldn't be affected by sizing,  I've tried it on all fonts and it's been fine.

Reply:
I agree, but `handleScroll` and the event listeners are not the proper solution to this problem. If you really want to solve it here are 2 possible solutions that comes to my mind rn:

- You can conditionally check if `scrollAmount` + `document.documentElement.scrollTop` exceeding the `maxScrollPos` you calculated and then update the scroll position based on that.
- If `scrollAmount` + `document.documentElement.scrollTop` colliding with the footer's Y position, then you can just scroll to the last multimedia container instead of updating scroll position with a static value.

Reply:
> If scrollAmount + document.documentElement.scrollTop colliding with the footer's Y position, then you can just scroll to the last multimedia container instead of updating scroll position with a static value.

updated based on this.